pkg/tfvars/ibmcloud/ibmcloud.go

```diff
@@ -96,26 +105,56 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 		Auth:                     sources.Auth,
 		BootstrapInstanceType:    masterConfig.Profile,
 		CISInstanceCRN:           sources.CISInstanceCRN,
+		ComputeSubnets:           workerSubnets,
+		ControlPlaneSubnets:      masterSubnets,
 		DNSInstanceID:            sources.DNSInstanceID,
+		EndpointsJSONFile:        sources.EndpointsJSONFile,
 		ImageFilePath:            cachedImage,
 		MasterAvailabilityZones:  masterAvailabilityZones,
 		MasterDedicatedHosts:     sources.MasterDedicatedHosts,
 		MasterInstanceType:       masterConfig.Profile,
 		NetworkResourceGroupName: sources.NetworkResourceGroupName,
+		PreexistingVPC:           sources.PreexistingVPC,
 		PublishStrategy:          string(sources.PublishStrategy),
 		Region:                   masterConfig.Region,
 		ResourceGroupName:        sources.ResourceGroupName,
-		WorkerAvailabilityZones:  workerAvailabilityZones,
-		WorkerDedicatedHosts:     sources.WorkerDedicatedHosts,
-		PreexistingVPC:           sources.PreexistingVPC,
 		VPC:                      vpc,
 		VPCPermitted:             sources.VPCPermitted,
-		ControlPlaneSubnets:      masterSubnets,
-		ComputeSubnets:           workerSubnets,
+		WorkerAvailabilityZones:  workerAvailabilityZones,
+		WorkerDedicatedHosts:     sources.WorkerDedicatedHosts,
 
 		// TODO: IBM: Future support
 		// ExtraTags:               masterConfig.Tags,
 	}
 
 	return json.MarshalIndent(cfg, "", "  ")
 }
+
+// CreateEndpointJSON creates JSON data containing IBM Cloud service endpoint override mappings.
+func CreateEndpointJSON(endpoints []configv1.IBMCloudServiceEndpoint, region string) ([]byte, error) {
+	// If no endpoint overrides, simply return
+	if len(endpoints) == 0 {
+		return nil, nil
+	}
+
+	// Following the format provided by the IBM Cloud Terraform Provider documentation:
+	// https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/guides/custom-service-endpoints#file-structure-for-endpoints-file
+	endpointContents := make(map[string]map[string]map[string]string)
```

Comment:
this seems like a fairly complex way of representing this data structure.  would it be possible to create a type representing it?

Reply:
Sure, I'll see about doing that instead, to get the JSON format correct.

Reply:
I made the design a bit better I hope.